simplejson/simplejson.go

```diff
@@ -0,0 +1,268 @@
+	}
+	return retval, nil
+}
+
+func (req *SimpleJsonQueryRequest) extractColumnValuesTSDB(column *pb.Column) (reflect.Value, string) {
+	colType := strings.ToLower(column.Dtype.String())
+	dataTypes := strings.Title(fmt.Sprintf("%ss", colType))
+	return reflect.Indirect(reflect.ValueOf(column)).FieldByName(dataTypes), colType
+}
+
+func (req *SimpleJsonQueryRequest) formatTimeTSDB(timestamp time.Time) interface{} {
+	return timestamp.UnixNano() / 1000000
+}
+func (req *SimpleJsonQueryRequest) getTargetTSDB(frame *pb.Frame) string {
+	labels := ""
+	metricName := ""
+	for _, column := range frame.Columns {
+		if column.Name != "values" {
+			values, _ := req.extractColumnValuesTSDB(column)
+			val := fmt.Sprintf("%s", values.Index(0).Interface())
+			if column.Name == "metric_name" {
+				metricName = val
+			} else {
+				labels += fmt.Sprintf("[%s=%s]", column.Name, val)
+			}
+		}
+	}
+	return metricName + labels
+}
+
+func (req *SimpleJsonQueryRequest) CreateResponse(iter frames.FrameIterator) (interface{}, error) {
+	formatters := map[string]func(iter frames.FrameIterator) (interface{}, error){"kv": req.formatKV, "tsdb": req.formatTSDB}
+	if formatter, ok := formatters[req.Backend]; ok {
+		return formatter(iter)
+	}
+	return nil, fmt.Errorf("Unknown format: %s", req.Backend)
+}
+
+func (req *SimpleJsonSearchRequest) CreateResponse(iter frames.FrameIterator) (interface{}, error) {
+	// Placeholder for actual implementation
+	return []string{}, nil
+}
+
+func (req *SimpleJsonQueryRequest) ParseRequest(requestBody []byte) error {
+	if err := json.Unmarshal(requestBody, req); err != nil {
+		return err
+	}
+	for _, target := range req.Targets {
+		req.Type = target["type"].(string)
+		fieldInput := target["target"].(string)
+		if err := req.parseQueryLine(fieldInput); err != nil {
+			return err
+		}
+	}
+	req.parseRange()
+	return nil
+}
+
+func (req *SimpleJsonSearchRequest) ParseRequest(requestBody []byte) error {
+	if err := json.Unmarshal(requestBody, req); err != nil {
+		return err
+	}
+	for _, target := range strings.Split(req.Target, QUERY_SEPARATOR) {
+		fmt.Println(target)
+		if err := req.parseQueryLine(strings.TrimSpace(target)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (req *SimpleJsonQueryRequest) parseQueryLine(fieldInput string) error {
+	translate := map[string]string{"table_name": "TableName"}
+	re := regexp.MustCompile(`^\s*(filter|fields|table_name|backend|container|username|password)\s*=\s*(.*)\s*$`)
+	for _, field := range strings.Split(fieldInput, QUERY_SEPARATOR) {
+		match := re.FindStringSubmatch(field)
+		var value interface{}
+		if len(match) > 0 {
+			fieldName := strings.Title(match[1])
+			if fieldNameTranslated, ok := translate[match[1]]; ok {
+				fieldName = fieldNameTranslated
+			}
+			if fieldName == "Fields" {
+				value = strings.Split(match[2], FIELDS_ITEMS_SEPARATOR)
+			} else {
+				value = match[2]
+			}
+			if err := setField(req, fieldName, value); err != nil {
```

Comment:
Not very idiomatic Go :) 
Can you give some example of query lines?

Reply:
I know, @tebeka , GO is not my "mother tongue", my background is python so it felt very natural to me to have a "setattr" LOL. The idea was to have a generic way to set the request fields once parsed from the query. Here is an example of such a query: `fields=sentiment;table_name=stock_metrics;backend=tsdb;filter=symbol==\"AAPL\";container=containerName;username=someuser;password=xyz`

Reply:
Yeah, coming from Python myself it takes a bit to get used to writing idiomatic Go. 
The input example should be a comment in the code - makes it easier to understand the parser.

Something like:
```go
// Parse a query line
// query example:
// fields=sentiment;table_name=stock_metrics;backend=tsdb;filter=symbol==\"AAPL\";container=containerName;username=someuser;password=xyz`
func parseQuery(queryLine string) (map[string]string, error) {
	query := make(map[string]string)
	for fNum, field := range strings.Split(queryLine, ";") {
		i := strings.Index(field, "=")
		if i == -1 {
			return nil, fmt.Errorf("%d: bad field = %s", fNum, field)
		}
		query[field[:i]] = field[i+1:]
	}

	return query, nil
}
```

Reply:
thanks for that, but I still need to inject the query into the simple json request, "Backend", "Fields".....